src/Features/Core/Portable/Diagnostics/CodeAnalysisDiagnosticAnalyzerService.cs

```diff
@@ -134,25 +134,41 @@ private async ValueTask AnalyzeProjectCoreAsync(Project project, Action<Project>
         }
 
         /// <summary>
-        /// Running code analysis on the project force computes and caches the diagnostics on the DiagnosticAnalyzerService.
-        /// We return these cached document diagnostics here, including both local and non-local document diagnostics.
+        /// Running code analysis on the project force computes and caches the diagnostics on the
+        /// DiagnosticAnalyzerService. We return these cached document diagnostics here, including both local and
+        /// non-local document diagnostics.
         /// </summary>
-        public Task<ImmutableArray<DiagnosticData>> GetLastComputedDocumentDiagnosticsAsync(DocumentId documentId, CancellationToken cancellationToken)
-            => _clearedProjectIds.Contains(documentId.ProjectId)
-                ? SpecializedTasks.EmptyImmutableArray<DiagnosticData>()
-                : _diagnosticAnalyzerService.GetCachedDiagnosticsAsync(_workspace, documentId.ProjectId,
-                    documentId, includeSuppressedDiagnostics: false, includeLocalDocumentDiagnostics: true,
-                    includeNonLocalDocumentDiagnostics: true, cancellationToken);
+        /// <remarks>
+        /// Only returns non-suppressed diagnostics.
+        /// </remarks>
+        public async Task<ImmutableArray<DiagnosticData>> GetLastComputedDocumentDiagnosticsAsync(DocumentId documentId, CancellationToken cancellationToken)
+        {
+            if (_clearedProjectIds.Contains(documentId.ProjectId))
+                return [];
+
+            var diagnostics = await _diagnosticAnalyzerService.GetCachedDiagnosticsAsync(
+                _workspace, documentId.ProjectId, documentId, includeLocalDocumentDiagnostics: true,
+                includeNonLocalDocumentDiagnostics: true, cancellationToken).ConfigureAwait(false);
+            return diagnostics.WhereAsArray(d => !d.IsSuppressed);
```

Comment:
any place that previously passed `includeSuppressedDiagnostics: false` into the service, now just filters the result.  

Note: i wouldn't mind having this not be here, and pushignt his even higher all the way to the feature layer.  but this is a simple starting step.